### PR TITLE
Drop support for Python 2.6

### LINKED ISSUE #1037: Drop support for Python 2.6
Remove any code needed to support Python 2.6, update docs, and remove package classifier.

### PR DESCRIPTION
**Pull request by hugovk**
Fixes #1037.

### CODE REVIEW

**Comment by hugovk** [comment]
The pep8 build job is also failing on latest master.

**Comment by GriffGeorge** [comment]
I fixed the pep8 job in #1133, if that's needed to merge this.

**Comment by hugovk** [comment]
@GriffGeorge Thanks, I've rebased on master and now Travis is green!

File: README.rst [hunk 1]
```diff
@@ -15,10 +15,9 @@ architectural style, and tries to do as little as possible while
 remaining highly effective.
 
 Falcon apps work with any WSGI server, and run great under
-CPython 2.6-2.7, PyPy 2.7, Jython 2.7, and CPython 3.3+.
+CPython 2.7, PyPy 2.7, Jython 2.7, and CPython 3.3+.
 
-(Note: Support for CPython 2.6 and Jython 2.7 is deprecated and will be
-removed in Falcon 2.0.)
+(Note: Support for Jython 2.7 is deprecated and will be removed in Falcon 2.0.)
```
Comment (cclauss) [inline README.rst]: Given that 2.7 is still the current version of Jython, I would recommend changing __Jython 2.7__ to __Jython__.

Comment (hugovk) [reply README.rst]: @cclauss Done!

**Comment by kgriffs** [comment]
Since this is a obviously a breaking change, we'll need to wait and put this in the next major revision (2.0). Once 1.4 is out the door, we'll pull this in for 2.0.

**Comment by hugovk** [comment]
What's the schedule for 1.4?

**Comment by kgriffs** [comment]
Hoping to have it out within the next 2 weeks.

**Comment by kgriffs** [comment]
Rebased and added a couple additional patches to turn travis green.

**Pull request merged**